File: src/intellibitz/sted/actions/DeleteAction.java
package intellibitz.sted.actions;

import intellibitz.sted.event.FontMapChangeEvent;
import intellibitz.sted.fontmap.FontMap;
import intellibitz.sted.ui.STEDWindow;

import javax.swing.*;
import javax.swing.event.ListSelectionEvent;
import java.awt.event.ActionEvent;
import java.util.Collection;

public class DeleteAction
        extends CutAction {
    public DeleteAction() {
        super();
    }


    public void valueChanged(ListSelectionEvent e) {
        final ListSelectionModel listSelectionModel =
                (ListSelectionModel) e.getSource();
        setEnabled(listSelectionModel.getMinSelectionIndex() >= 0);
    }

    public void actionPerformed(ActionEvent e) {
        final FontMap fontMap =
                getSTEDWindow().getDesktop()
                        .getFontMap();
        final Collection entries = delete();
        pushUndo(entries, fontMap.getEntries().getUndo());
        fontMap.setDirty(!entries.isEmpty());
        fontMap.fireUndoEvent();
        fireStatusPosted("Deleted");
    }

    private Collection delete() {
        final STEDWindow stedWindow = getSTEDWindow();
        final int result = JOptionPane.showConfirmDialog
                (stedWindow, "Do you want to delete selected row(s)", "confirm",
                        JOptionPane.OK_CANCEL_OPTION);
        if (result == JOptionPane.OK_OPTION) {
            return cut();
        }
        return null;
    }


    public void stateChanged(FontMapChangeEvent e) {
        setEnabled(!getSelectedRows().isEmpty());
    }

}

File: src/main/intellibitz/sted/actions/DeleteAction.kt
package sted.actions

import sted.event.FontMapChangeEvent
import java.awt.event.ActionEvent
import javax.swing.JOptionPane
import javax.swing.ListSelectionModel
import javax.swing.event.ListSelectionEvent

class DeleteAction : CutAction() {
    override fun valueChanged(e: ListSelectionEvent) {
        val listSelectionModel = e.source as ListSelectionModel
        isEnabled = listSelectionModel.minSelectionIndex >= 0
    }

    override fun actionPerformed(e: ActionEvent) {
        val fontMap = stedWindow.desktop.fontMap
        val entries = delete()
        pushUndo(entries!!, fontMap.entries.undo)
        fontMap.isDirty = !entries.isEmpty()
        fontMap.fireUndoEvent()
        fireStatusPosted("Deleted")
    }

    private fun delete(): Collection<*>? {
        val result = JOptionPane.showConfirmDialog(
            stedWindow, "Do you want to delete selected row(s)", "confirm",
            JOptionPane.OK_CANCEL_OPTION
        )
        return if (result == JOptionPane.OK_OPTION) {
            cut()
        } else null
    }

    override fun stateChanged(e: FontMapChangeEvent) {
        isEnabled = !selectedRows.isEmpty()
    }
}